File: java/Aluno.java
public class Aluno{
   private String nome;
   private String email;

   public void aluno(String nome){
    this.nome = nome;
    this.email = gerarEmail(nome);
   }


    private String gerarEmail(String email){
        String [] partes = nome.toLowerCase().split(" ");
        if(partes.length > 1){
        return partes[0] + "." + partes[partes.length - 1] + "@ufn.edu.br";
        }
        return partes[0] + "@ufn.edu.br";
    }
    //gettter e setter para acessar dados privados
    public String getNome(){return nome;}
    public String getEmail(){return email;}

    public void setNome(String nome){
        this.nome = nome;
        this.email = gerarEmail(nome);
    }
    public void setEmail(String email){this.email = email;}

    @Override
    public String toString(){
        return "Aluno: " + nome + "email: " + email;
    }
}

File: python/Desafio/Aluno.py
class Aluno:
    def __init__(self, nome):
        """Inicializa um objeto Aluno com nome e email gerado automaticamente.

        Args:
            nome (str): Nome completo do aluno.
        """
        self.nome = nome;
        self.email = self.gerarEmail(nome)

    def gerarEmail(self, email):
        """Gera um email institucional a partir do nome do aluno.

        Args:
            email (str): Nome completo do aluno usado para gerar o email.

        Returns:
            str: Email no formato primeiro.ultimo@ufn.edu.br.
        """
        partes = nome.lower().split()
        return partes[0] + "." + partes[-1] + "@ufn.edu.br"



def cadastrarAlunos(listaDeAlunos):
    """Cadastra 3 alunos via input do usuário e os adiciona à lista.

    Args:
        listaDeAlunos (list): Lista onde os objetos Aluno serão inseridos.
    """
    for i in range(3):
        nome = input("Insira o nome do aluno: ")
        aluno = Aluno(nome)  # cria novo aluno
        listaDeAlunos.append(aluno)
